src/Psalm/Internal/Analyzer/Statements/Expression/Assignment/PropertyAssignmentAnalyzer.php

```diff
@@ -237,6 +238,20 @@ public static function analyzeInstance(
                     continue;
                 }
 
+                if ($lhs_type_part instanceof Type\Atomic\TTemplateParam) {
+                    $extra_types = $lhs_type_part->extra_types;
+
+                    $lhs_type_part = array_values(
+                        $lhs_type_part->as->getAtomicTypes()
+                    )[0];
```

Comment:
Is this guaranteed to always have at lest one element?

Reply:
Yes. `getAtomicTypes()` returns `non-empty-array`, `array_values()` turn that into `non-empty-list`, so there's always an element at offset `0`.